### PR TITLE
#15517 Migrate Chat GPT endpoint to package

### PR DESCRIPTION
**Pull request by pasaperez-slingr**
Pull-request for issue https://github.com/slingr-stack/platform/issues/15517 created by @pasaperez-slingr

## What it does?

Creation of the chat gpt api package. Equivalence to what would be the endpoint.

## How to test it?

Register the package on developer portal and from one app install and make some request according to the api.

## Implementation notes

Create a the tag v2.0.0 and push.

## Deployment notes

Register on local, idea2, tarzan and later on prod.

### CODE REVIEW

File: scripts/api.js [hunk 1]
```diff
@@ -288,55 +253,33 @@ let Skeleton = function (options) {
...
-    sys.logs.debug('[skeleton] Setting header token oauth');
+function setAuthorization(options) {
+    sys.logs.debug('[chatgpt] Setting header token oauth');
     let authorization = options.authorization || {};
     authorization = mergeJSON(authorization, {
         type: "oauth2",
-        accessToken: sys.storage.get(config.get("oauth").id + ' - access_token', {decrypt:true}),
-        headerPrefix: "token"
+        accessToken: config.get("apiToken"),
+        headerPrefix: "Bearer"
```
Comment (agreggio-slingr) [inline scripts/api.js:277]: I see that you are using OAuth2 authentication, but the implementation is not present in the package.json file

**Review by agreggio-slingr — COMMENTED** [review]
Did you run the tests? Do you have credentials to share? How can I test the package if a tag hasn't been created yet?

File: scripts/api.js (hunk 1 above)
Comment (pasaperez-slingr) [reply scripts/api.js:277]: I only use the oauth parameter to add the authorization header. No extra implementation is needed.

**Comment by pasaperez-slingr** [comment]
> Did you run the tests? Do you have credentials to share? How can I test the package if a tag hasn't been created yet?

ChatGPT uses the same implementation than Dall-E, are only separated to differentiate the services. You can test with Dall-E package.

**Review by agreggio-slingr — APPROVED** [review]

**Review by agreggio-slingr — APPROVED** [review]

**Pull request merged**